### PR TITLE
rpc/client: provide scripts in AddNetworkFee

### LINKED ISSUE #1304: Add RPC client NEP5 functionality tests
It has around zero coverage at the moment.

### PR DESCRIPTION
**Pull request by fyrchik**
To calculate network fee properly we must know type of every
signer (simple, multisig, contract). Providing scripts is the most
simple and flexible way to know this.

Also add tests for some functions from `client` package.
Close #1304.

### CODE REVIEW

File: pkg/rpc/client/rpc.go [hunk 1]
```diff
@@ -513,27 +512,23 @@ func (c *Client) CalculateValidUntilBlock() (uint32, error) {
 }
 
 // AddNetworkFee adds network fee for each witness script and optional extra
-// network fee to transaction.
-func (c *Client) AddNetworkFee(tx *transaction.Transaction, extraFee int64, acc *wallet.Account) error {
+// network fee to transaction. Scripts is an array verification signer's verification scripts.
+// nil means that signer is a contract, otherwise it must be either simple or multisig script.
+func (c *Client) AddNetworkFee(tx *transaction.Transaction, extraFee int64, scripts ...[]byte) error {
```
Comment (roman-khimov) [inline pkg/rpc/client/rpc.go:497]: I'm really unsure it improves anything. Theoretically, you could use `Contract.Parameters` to estimate invocation script size even for non-standard contracts, but changing API this way you lose this ability.

File: pkg/rpc/client/rpc.go [hunk 2]
```diff
@@ -513,27 +512,23 @@ func (c *Client) CalculateValidUntilBlock() (uint32, error) {
...
-	if acc.Contract != nil {
-		netFee, sizeDelta := core.CalculateNetworkFee(acc.Contract.Script)
-		tx.NetworkFee += netFee
-		size += sizeDelta
-	}
-	for _, cosigner := range tx.Signers {
-		script := acc.Contract.Script
-		if !cosigner.Account.Equals(hash.Hash160(acc.Contract.Script)) {
+	for i, cosigner := range tx.Signers {
+		if scripts[i] == nil {
```
Comment (AnnaShaleva) [inline pkg/rpc/client/rpc.go:500]: Scripts length might be less than the length of tx.Signers.

File: pkg/rpc/client/rpc.go [hunk 3]
```diff
@@ -513,27 +512,23 @@ func (c *Client) CalculateValidUntilBlock() (uint32, error) {
 }
 
 // AddNetworkFee adds network fee for each witness script and optional extra
-// network fee to transaction.
-func (c *Client) AddNetworkFee(tx *transaction.Transaction, extraFee int64, acc *wallet.Account) error {
+// network fee to transaction. Scripts is an array verification signer's verification scripts.
```
Comment (AnnaShaleva) [inline pkg/rpc/client/rpc.go:495]: Typo in `verification signer's verification`.

File: pkg/rpc/client/rpc.go (hunk 2 above)
Comment (fyrchik) [reply pkg/rpc/client/rpc.go:500]: Fixed, added test.

File: pkg/rpc/client/rpc.go (hunk 1 above)
Comment (fyrchik) [reply pkg/rpc/client/rpc.go]: Replaced them with accounts.

**Pull request merged**